components/raftstore/src/store/peer.rs

```diff
@@ -390,7 +429,10 @@ impl<S: Snapshot> CmdEpochChecker<S> {
             .rev()
             .find(|cmd| {
                 (check_ver && cmd.epoch_state.change_ver)
-                    || (check_conf_ver && cmd.epoch_state.change_conf_ver)
+                    || (check_conf_ver
+                        // If it's a mutually exclusive command, it should be checked with all
+                        // commands which change the `conf_ver`.
+                        && (cmd.epoch_state.change_conf_ver || cmd.is_mutually_exclusive()))
```

Comment:
It seems we allow: propose split -> propose transfer leader
but prohibit: propose transfer leader -> propose split ?

Reply:
Nope, both forbid.

`Split` will change the `conf_ver`, so if there exists pending `Split` commands in the `proposed_admin_cmd` lists, the `TransferLeader` will not be passed through by the function `last_conflict_index(...)`, returning the index of the pending `Split` command.